### PR TITLE
Upload static media to s3 on deploy

### PR DESCRIPTION
**Pull request by pmac**
This copies the files out of the docker container and uploads them to the stage and prod s3 buckets when a prod deploy is happening (stage and prod are both deployed during a prod deployment). 

/cc @jpetto @jgmize @metadave

### CODE REVIEW

File: bin/upload-staticfiles.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+for BUCKET in stage prod; do
+    # hashed filenames
+    aws s3 sync \
+        --exclude "*" \
+        --include "*.????????????.*" \
+        --acl public-read \
+        --cache-control "max-age=315360000, public, immutable" \
+        --profile bedrock-media \
+        . "s3://bedrock-${BUCKET}-media/"
+    # non-hashed filenames
```
Comment (metadave) [inline bin/upload-staticfiles.sh]: are lines 30 and 31 the same comment repeated?

File: bin/upload-staticfiles.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    # may not need to include non-hashed files
+    # TODO look into this if this is slow or makes the bucket too large
```
Comment (metadave) [inline bin/upload-staticfiles.sh]: mdn syncs required an [additional flag](https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html) `--page-size` to help prevent timeouts on large buckets.

File: bin/upload-staticfiles.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+cd static
+for BUCKET in stage prod; do
+    # hashed filenames
+    aws s3 sync \
+        --exclude "*" \
+        --include "*.????????????.*" \
+        --acl public-read \
+        --cache-control "max-age=315360000, public, immutable" \
+        --profile bedrock-media \
```
Comment (metadave) [inline bin/upload-staticfiles.sh]: any reason for using `--profile` vs the `AWS_PROFILE` env var?

Comment (pmac) [reply bin/upload-staticfiles.sh]: that I didn't know about `AWS_PROFILE` and was just copying what IRL did:

https://github.com/mozilla/irlpodcast/blob/91b8bbb35da7f9ff3b1e844b7ce103bfd43eca0a/Jenkinsfile#L17

File: bin/upload-staticfiles.sh (hunk 2 above)
Comment (pmac) [reply bin/upload-staticfiles.sh]: according to the docs:

```
--page-size (integer) The number of results to return in each response to a list operation. The default value is 1000 (the maximum allowed). Using a lower value may help if an operation times out.
```

To my knowledge we're not doing any list operations. Or is it doing that behind-the-scenes during the sync?

File: bin/upload-staticfiles.sh (hunk 3 above)
Comment (pmac) [reply bin/upload-staticfiles.sh]: Happy to switch it if one is better than the other though.

File: bin/upload-staticfiles.sh (hunk 2 above)
Comment (metadave) [reply bin/upload-staticfiles.sh]: I seem to remember troubleshooting the MDN sync and seeing that `aws s3 sync` used list to compare files.

File: bin/upload-staticfiles.sh (hunk 1 above)
Comment (pmac) [reply bin/upload-staticfiles.sh]: They look different to me locally. Line 30 says what the next section is doing, and 31 says that we may not eventually need to do this and 32 says we should look into that someday if it's a problem.

**Comment by pmac** [comment]
The buckets are now populated as @jpetto and I did a test run of this:

https://ci.us-west.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/static-media-to-s3/1/pipeline

File: bin/upload-staticfiles.sh (hunk 2 above)
Comment (pmac) [reply bin/upload-staticfiles.sh]: @metadave do you recall to what value they set `--page-size`?

Comment (metadave) [reply bin/upload-staticfiles.sh]: [100](https://github.com/mdn/infra/blob/master/apps/mdn/utils/mdn_backup_cron/image/mdn_sync.sh#L11)

Comment (metadave) [reply bin/upload-staticfiles.sh]: definitely not required, but keep it in mind in case there are sync issues.

**Comment by pmac** [comment]
To be clear: what this PR does is upload the media to the buckets during deployment. It does not change the app in any way to make it use the s3 buckets for serving prod assets. My proposal is:

1. Merge this PR
2. Deploy production
3. Change settings in CloudFlare for staging to point `/media/*` to the staging s3 bucket
4. Test staging extensively to make sure this works as expected
5. Make the same changes in CloudFlare for prod
6. Profit! 💰💰💰💰

**Pull request merged**